lib/chef/application/apply.rb

```diff
@@ -148,6 +154,10 @@ def read_recipe_file(file_name)
     if file_name.nil?
       Chef::Application.fatal!("No recipe file was provided", Chef::Exceptions::RecipeNotFound.new)
     else
+      if file_name =~ /\.yml$/
```

Comment:
```suggestion
      if File.extname(file_name) == ".yml"
```

Should this use `File.extname` to avoid any potential regex weirdness?

Reply:
seems good, I was cargo culting.